src/schema/v2/artist/targetSupply/__tests__/index.test.ts

```diff
@@ -0,0 +1,266 @@
+          },
+          artworks: [
+            {
+              artwork: {
+                internalID: "5e3854df108c9200113cd354",
+              },
+              realizedPrice: "$41,595",
+            },
+            {
+              artwork: {
+                internalID: "5e3854d523363e000f352f7f",
+              },
+              realizedPrice: "$36,395",
+            },
+            {
+              artwork: {
+                internalID: "5e3854cf20273e00122131a5",
+              },
+              realizedPrice: "$4,549",
+            },
+            {
+              artwork: {
+                internalID: "5e3854ce662102000e73328f",
+              },
+              realizedPrice: "$20,797",
+            },
+            {
+              artwork: {
+                internalID: "59d6a884c9dc240f49917380",
+              },
+              realizedPrice: "$1,850",
+            },
+            {
+              artwork: {
+                internalID: "5de86d009a893d001244f5fe",
+              },
+              realizedPrice: "$2,500",
+            },
+            {
+              artwork: {
+                internalID: "5dcdc5a4f78568000ede76dc",
+              },
+              realizedPrice: "$325",
+            },
+          ],
+        },
+      })
+    })
+
+    it("randomizes artworks if `randomize` argument is passed", async () => {
+      const query = `
+        {
+          artist(id:"alex-katz") {
+            targetSupply {
+              microfunnel {
+                artworks(randomize: true) {
+                  artwork {
+                    internalID
+                  }
+                  realizedPrice
+                }
+              }
+            }
+          }
+        }
+      `
+      const context = {
+        artistLoader: () => {
+          return Promise.resolve({
+            id: "banksy",
+            name: "Banksy",
+          })
+        },
+        artworksLoader: () => {
+          return Promise.resolve([
+            {
+              _id: "5e3854df108c9200113cd354",
+            },
+            {
+              _id: "5e3854d523363e000f352f7f",
+            },
+            {
+              _id: "5e3854cf20273e00122131a5",
+            },
+            {
+              _id: "5e3854ce662102000e73328f",
+            },
+            {
+              _id: "59d6a884c9dc240f49917380",
+            },
+            {
+              _id: "5de86d009a893d001244f5fe",
+            },
+            {
+              _id: "5dcdc5a4f78568000ede76dc",
+            },
+          ])
+        },
+      }
+      const responseA = await runQuery(query, context)
+      const responseB = await runQuery(query, context)
+      const idsA = responseA.artist.targetSupply.microfunnel.artworks.map(
+        ({ artwork }) => artwork.internalID
+      )
+      const idsB = responseB.artist.targetSupply.microfunnel.artworks.map(
+        ({ artwork }) => artwork.internalID
+      )
+      expect(idsA).not.toEqual(idsB)
```

Comment:
What I think this test is doing is running the query twice and then asserting that the ids do not come back in the same order, right? I think this means that it's possible for this test to be flaky and randomly go red when it just so happens that two shuffles in a row result in the same artworks. But then this flakiness is mitigated by the fact that there are 7 artworks in play and so the odds of the shuffles matching are just really low. Is that right?

Reply:
Yup, if we start getting flake at this test we can modify, but it's pretty unlikely that this will return two identical rolls back to back. If it _does_ happen however, the test description and the content points immediately at the issue.

Reply:
I happened to be talking to my physicist brother and asked him to help me compute the odds - it's 0.000198413 (1/5040) so we're probably good, haha! 😝